backend/app/crud/crud_role.py

```diff
@@ -37,25 +37,20 @@ async def get_by_name(self, db, name: str) -> Role | None:
 
     async def create(self, db, obj_in: CreateRole) -> NoReturn:
         new_role = self.model(**obj_in.dict(exclude={'menus'}))
-        menu_list = []
-        for menu_id in obj_in.menus:
-            menu_list.append(await db.get(Menu, menu_id))
-        new_role.menus.append(*menu_list)
+        menus = await db.execute(select(Menu).where(Menu.id.in_(obj_in.menus)))
+        new_role.menus = menus.scalars().all()
         db.add(new_role)
 
     async def update(self, db, role_id: int, obj_in: UpdateRole) -> int:
         role = await db.execute(
-            update(self.model).where(self.model.id == role_id).values(**obj_in.dict(exclude={'menus'}))
+            update(self.model)
+            .where(self.model.id == role_id)
+            .values(**obj_in.dict(exclude={'menus'}))
         )
         current_role = await self.get_with_relation(db, role_id)
-        # 删除角色所有菜单
-        for i in list(current_role.menus):
-            current_role.menus.remove(i)
```

Comment:
remove()

https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#deleting-rows-from-the-many-to-many-table